### DISCUSSION TITLE
Right click measurement show actual trace values

### DISCUSSION THREAD

**Discussion by jsiddall** [OpenHantek/OpenHantek6022#332] [Ideas]
Currently the right mouse button measurement mode will show you the current _mouse_ position on the scale of a given trace(s) when you are suitably close, but never shows the trace values.  Far more useful would be to show the actual _trace_ value(s) if they are near the cursor.  Ideally the data point being displayed would be highlighted with a dot to indicate visually which value is being shown in the data display.  Here is an example screenshot to help explain:

![OpenHantek6022MeasureMockup](https://user-images.githubusercontent.com/7401394/198080151-5a830a38-7580-4935-b4ba-605545878450.png)

Note that the values shown in the pop-up measurement window are the values at the yellow/blue dots on the traces, not the position of the mouse itself.

**Comment by Ho-Ro** [comment]
During development of this right mouse function I also thought about this kind of display, but refused to follow this way.
1. If the signal is not stationary you will get quickly changing numbers - difficult to read.
2. Often the real value of the current trace is not important, but you want to check a limit.
3. And last but not least it is easy to retrieve the mouse position on the scope frame, but it is more difficult to get the trace value because this would violate the data encapsulation.

**Comment by jsiddall** [comment]
Yes, I understand there are some things to consider.  For the points above:
1.  This is true, and this feature probably makes the most sense when paused, but a couple of ways to address this for continuous updates would be to show the value at the time you right click and don't update it until you right click again, or update it slowly (ex: once per second).
2. I'm not sure what you are saying, but if a user wanted to measure it then the current trace must be relevant  
3. I believe you!  But it would still be nice to be able to know the value at any point in the trace.

One other option I considered, which would provide similar measurement benefits, would be to use the cursors as the time/frequency values.  In fact, this is basically how my old Tektronix scope works: it shows the trace values where they intersects the two cursors.  The benefit of this is that it could possibly be added to where cursor data is displayed and wouldn't require any interactions with the mouse directly.

**Reply by Ho-Ro** [reply]
> I'm not sure what you are saying, but if a user wanted to measure it then the current trace must be relevant

What I want to say is if I measure a logic signal then I want to see if the trace is above the min H level of e.g. 2V or below the max L level of 0.8 V, it doesn't matter (IMHO) if the value is 1.2 V or 1.5 V - it shows me that something went wrong.

**Reply by jsiddall** [reply]
Got it.  Yes, for digital signals that makes sense.  To that end, a feature that added dotted user-configurable threshold lines would _also_ be nice!
